example-environment.lua

```diff
@@ -191,7 +205,7 @@ E.p_bad:on_occupancy(function()
   ctx.task(function()
     ctx:fire{"Env", E.p_bad}
     E.l_bad:switch"On"
-    local ok = ctx:wait({"Env", E.p_bad}, nil, 300)
+    ok = ctx:wait({"Env", E.p_bad}, nil, 300)
```

Comment:
hm. No, it ought to be local, not a global, I think?

Reply:
luacheck warns about shadowing upvalue:
`example-environment.lua:206:11: shadowing upvalue ok on line 201`. Looks like we can use already defined var in upper function.

Reply:
nah, the shadowing is exactly right here. It would be a bit awkward to have a closure for an outer "ok". This is why I have a love-hate-relationship towards luacheck :-) ... I guess it's easiest to just rename either inner or outer "ok".

Reply:
```lua
index 82ee918..3b8ddbe 100755
--- a/example-environment.lua
+++ b/example-environment.lua
@@ -203,8 +203,8 @@ E.p_bad:on_occupancy(function()
   ctx.task(function()
     ctx:fire{"Env", E.p_bad}
     E.l_bad:switch"On"
-    ok = ctx:wait({"Env", E.p_bad}, nil, 300)
-    if not ok then
+    local res = ctx:wait({"Env", E.p_bad}, nil, 300)
+    if not res then
       E.l_bad:switch"Off"
     end
   end)
```